### DISCUSSION TITLE
MF6 not running through Notebook -- os.system() issue?

### DISCUSSION THREAD

**Discussion by ah-pruitt** [gmdsi/GMDSI_notebooks#135] [General]
When I try to run the Freyberg model with pyemu, I'm getting an exception for a non-zero return in the notebook. But in the anaconda prompt, it looks like it starts with a "system cannot find the path specified", but then runs the simulation anyways with a "normal termination of simulation". (Screenshots attached).
 
![pyemu_exception](https://github.com/user-attachments/assets/e31862ff-6db7-493e-af6b-7a5e6d9a0633)
![conda_runMF6anyways](https://github.com/user-attachments/assets/632227a2-9340-4a71-8c10-1afa03e6a2c5)

This looks like it's an issue with the os.system() command in the pyemu utilities. I have the same issue if I try to manually feed the command to os.system() to run mf6 in the sim_ws.

Windows machine, using the GMDSITUT environment. It's a governmental laptop, so could technically be something related to read/write permissions, but wanted to see if it was something on the python side first. I'm not sure what file it's not finding if it's running the mf6.exe anyways. 

Anyone else have this issue or suggestions?

**Comment by jtwhite79** [comment]
@ah-pruitt can you post your model files dir that was giving this error?

**Reply by ah-pruitt** [reply]
I'm in a clean clone of the basic Freyberg model notebook ('/tutorials/part0_02_intro_to_freyberg_model'). But I was having similar issues in the pstfrom notebook. 

Looks like mf6 will run using subprocess.run with a return value of 0. Maybe os.system is finally dead for windows? But that means scripts will die anytime it calls the pyemu os.utils. 

![mf6_subprocess](https://github.com/user-attachments/assets/6f8df114-be0b-4429-945c-58ee2a157692)

**Reply by jtwhite79** [reply]
Do you have mf6 in your path so that maybe a diff version of mf6 is being invoked?  The notebooks are continually tested on Linux and Windows (mostly) without issues so I'd like to see your dir.   I think you can just paste the dir here on gh...

**Reply by jtwhite79** [reply]
And you are using the tutorial env?  Just checking...

**Reply by ah-pruitt** [reply]
Assuming you just want the list of files in the working directory, results from os.listdir() for the sim_ws directory:

['freyberg6.dis',
 'freyberg6.dis.grb',
 'freyberg6.dis_botm_layer1.txt',
 'freyberg6.dis_delc.txt',
 'freyberg6.dis_delr.txt',
 'freyberg6.dis_idomain_layer1.txt',
 'freyberg6.dis_top.txt',
 'freyberg6.ghb',
 'freyberg6.ghb_stress_period_data_1.txt',
 'freyberg6.gwfgwt',
 'freyberg6.ic',
 'freyberg6.ic_conc_strt_layer1.txt',
 'freyberg6.ic_strt_layer1.txt',
 'freyberg6.ims',
 'freyberg6.lst',
 'freyberg6.nam',
 'freyberg6.npf',
 'freyberg6.npf_icelltype_layer1.txt',
 'freyberg6.npf_k33_layer1.txt',
 'freyberg6.npf_k_layer1.txt',
 'freyberg6.obs',
 'freyberg6.oc',
 'freyberg6.rch',
 'freyberg6.rch_recharge_1.txt',
 'freyberg6.rch_recharge_10.txt',
 'freyberg6.rch_recharge_11.txt',
 'freyberg6.rch_recharge_12.txt',
 'freyberg6.rch_recharge_13.txt',
 'freyberg6.rch_recharge_14.txt',
 'freyberg6.rch_recharge_15.txt',
 'freyberg6.rch_recharge_16.txt',
 'freyberg6.rch_recharge_17.txt',
 'freyberg6.rch_recharge_18.txt',
 'freyberg6.rch_recharge_19.txt',
 'freyberg6.rch_recharge_2.txt',
 'freyberg6.rch_recharge_20.txt',
 'freyberg6.rch_recharge_21.txt',
 'freyberg6.rch_recharge_22.txt',
 'freyberg6.rch_recharge_23.txt',
 'freyberg6.rch_recharge_24.txt',
 'freyberg6.rch_recharge_25.txt',
 'freyberg6.rch_recharge_3.txt',
 'freyberg6.rch_recharge_4.txt',
 'freyberg6.rch_recharge_5.txt',
 'freyberg6.rch_recharge_6.txt',
 'freyberg6.rch_recharge_7.txt',
 'freyberg6.rch_recharge_8.txt',
 'freyberg6.rch_recharge_9.txt',
 'freyberg6.sfr',
 'freyberg6.sfr_connectiondata.txt',
 'freyberg6.sfr_packagedata.txt',
 'freyberg6.sfr_perioddata_1.txt',
 'freyberg6.sft_package_data.txt',
 'freyberg6.sto',
 'freyberg6.sto_iconvert_layer1.txt',
 'freyberg6.sto_ss_layer1.txt',
 'freyberg6.sto_sy_layer1.txt',
 'freyberg6.tdis',
 'freyberg6.wel',
 'freyberg6.wel_stress_period_data_1.txt',
 'freyberg6.wel_stress_period_data_10.txt',
 'freyberg6.wel_stress_period_data_11.txt',
 'freyberg6.wel_stress_period_data_12.txt',
 'freyberg6.wel_stress_period_data_13.txt',
 'freyberg6.wel_stress_period_data_14.txt',
 'freyberg6.wel_stress_period_data_15.txt',
 'freyberg6.wel_stress_period_data_16.txt',
 'freyberg6.wel_stress_period_data_17.txt',
 'freyberg6.wel_stress_period_data_18.txt',
 'freyberg6.wel_stress_period_data_19.txt',
 'freyberg6.wel_stress_period_data_2.txt',
 'freyberg6.wel_stress_period_data_20.txt',
 'freyberg6.wel_stress_period_data_21.txt',
 'freyberg6.wel_stress_period_data_22.txt',
 'freyberg6.wel_stress_period_data_23.txt',
 'freyberg6.wel_stress_period_data_24.txt',
 'freyberg6.wel_stress_period_data_25.txt',
 'freyberg6.wel_stress_period_data_3.txt',
 'freyberg6.wel_stress_period_data_4.txt',
 'freyberg6.wel_stress_period_data_5.txt',
 'freyberg6.wel_stress_period_data_6.txt',
 'freyberg6.wel_stress_period_data_7.txt',
 'freyberg6.wel_stress_period_data_8.txt',
 'freyberg6.wel_stress_period_data_9.txt',
 'freyberg6_freyberg.cbc',
 'freyberg6_freyberg.hds',
 'freyberg_mp.mpbas',
 'freyberg_mp.mpnam',
 'freyberg_mp.mpsim',
 'freyberg_mp.ne_layer1.txt',
 'freyberg_mp.ne_layer2.txt',
 'freyberg_mp.ne_layer3.txt',
 'heads.csv',
 'inschek.exe',
 'mf5to6.exe',
 'mf6.exe',
 'mfsim.lst',
 'mfsim.nam',
 'mp7.exe',
 'pestchek.exe',
 'pestpp-da.exe',
 'pestpp-glm.exe',
 'pestpp-ies.exe',
 'pestpp-mou.exe',
 'pestpp-opt.exe',
 'pestpp-sen.exe',
 'pestpp-swp.exe',
 'pm.pg1.sloc',
 'sfr.csv',
 'sfr.obs',
 'sim_ws.csv',
 'tempchek.exe',
 'zbud6.exe']

**Reply by ah-pruitt** [reply]
And yeah, using the gmdsitut environment.

**Reply by jtwhite79** [reply]
Actually I was hoping for the files so I could take then to a clean win vm and test a few things

**Reply by ah-pruitt** [reply]
[freyberg_mf6.zip](https://github.com/user-attachments/files/16420079/freyberg_mf6.zip)
Had to .zip it - git won't support all the filetypes. Let me know if you need it a different way.

**Reply by jtwhite79** [reply]
well I cant trigger the exception on a clean win11 vm with a fresh install of the gmdsitut env.  Can you post that notebook you show some screen shots from?

**Reply by ah-pruitt** [reply]
[intro_freyberg_model_OSerr.zip](https://github.com/user-attachments/files/16432713/intro_freyberg_model_OSerr.zip)
Again, zipped so gh can accept it. 

Running from the GMDSI_notebooks\tutorials\part0_02_intro_to_freyberg_model folder. Only real changes from the original notebook right now are adding a cell to use the sim.run_simulation and a cell to use subprocess to call mf6.
